### PR TITLE
checker: fix unhealth region skip the rule check

### PR DESCRIPTION
**Pull request by nolouch**
### What problem does this PR solve?

Issue Number: Close https://github.com/tikv/pd/issues/6426

### What is changed and how does it work?

```commit-message
allow the `schedule=deny` label can do rule constraints check
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: server/cluster/cluster.go [hunk 1]
```diff
@@ -70,6 +70,7 @@ var (
 	// WithLabelValues is a heavy operation, define variable to avoid call it every time.
 	regionUpdateCacheEventCounter = regionEventCounter.WithLabelValues("update_cache")
 	regionUpdateKVEventCounter    = regionEventCounter.WithLabelValues("update_kv")
+	regionSkipSchedulerCounter    = regionEventCounter.WithLabelValues("region_skip_scheduler")
```
Comment (rleungx) [inline server/cluster/cluster.go:73]: How about region_schedule_disabled?

File: pkg/schedule/operator_controller.go [hunk 2]
```diff
@@ -424,14 +423,6 @@ func (oc *OperatorController) checkAddOperator(isPromoting bool, ops ...*operato
 		if op.SchedulerKind() == operator.OpAdmin || op.IsLeaveJointStateOperator() {
 			continue
 		}
-		if cl, ok := oc.cluster.(interface{ GetRegionLabeler() *labeler.RegionLabeler }); ok {
```
Comment (rleungx) [inline pkg/schedule/operator_controller.go:427]: It seems we need to use the way of pausing schedulers to prevent the range from being scheduled.

Comment (nolouch) [reply pkg/schedule/operator_controller.go:427]: I add the tests, prevent it in `coordinator`.

**Review by rleungx — APPROVED** [review]

**Review by rleungx — APPROVED** [review]

File: pkg/schedule/checker/checker_controller.go [hunk 3]
```diff
@@ -125,6 +120,14 @@ func (c *Controller) CheckRegion(region *core.RegionInfo) []*operator.Operator {
 			c.regionWaitingList.Put(region.GetID(), nil)
 		}
 	}
+	// only consider merge checker when region label is set to "schedule=deny".
+	if cl, ok := c.cluster.(interface{ GetRegionLabeler() *labeler.RegionLabeler }); ok {
```
Comment (bufferflies) [inline pkg/schedule/checker/checker_controller.go:124]: maybe putting it into code line 137  be better?

Comment (nolouch) [reply pkg/schedule/checker/checker_controller.go:125]: This is not just for merge, but only merge is left. I change the comments.

File: server/cluster/coordinator.go [hunk 4]
```diff
@@ -901,7 +901,17 @@ func (s *scheduleController) Schedule(diagnosable bool) []*operator.Operator {
...
+		for _, op := range ops {
+			if labelMgr := s.cluster.GetRegionLabeler(); labelMgr != nil {
+				if labelMgr.ScheduleDisabled(s.cluster.GetRegion(op.RegionID())) {
+					denySchedulersByLabelerCounter.Inc()
+					foundDisabled = true
+					break
+				}
+			}
+		}
+		if len(ops) > 0 && !foundDisabled {
```
Comment (bufferflies) [inline server/cluster/coordinator.go:914]: In the origin logical code, the next interval should be min interval even if the scheduled region is disabled scheduled.

**Review by bufferflies — APPROVED** [review]
rest LGTM

**Pull request merged**